flik/linesearch/test/test_conditions.py

```diff
@@ -1 +1,35 @@
 """Tests for flik.linesearch.conditions."""
+
+import numpy as np
+from nose.tools import assert_raises
+
+from conditions import check_armijo
+
+def test_armijo_pars():
+    """
+    Test for parameters in Armijo's condition
+    """
+    def fsq(var):
+        "Little function to square a vector"
+        return var**2
```

Comment:
The function must be a scalar-valued function. If `var` is an array, `fsq` will return an array, which isn't compatible with Armijo condition. I think there might be generalization of Armijo (and line search in general) to multiple dimension, but it isn't quite nice.

In this case, `np.sum(var**2)` will return a scalar value (and you can use the same gradient function)